request-manager/spec/common_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020, Square, Inc.
+
+package spec
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/go-test/deep"
+)
+
+const (
+	specsDir = "../test/specs/"
+)
+
+var (
+	printf = func(s string, args ...interface{}) { fmt.Printf(s, args...) }
+)
+
+func runSequenceCheckOn(check SequenceCheck, filename string) error {
+	filepath := specsDir + filename
+	allSpecs, err := ParseSpec(filepath, printf)
+	if err != nil {
+		return err
+	}
+	for _, sequence := range allSpecs.Sequences {
+		if err := check.CheckSequence(*sequence); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func runNodeCheckOn(check NodeCheck, filename string) error {
+	filepath := specsDir + filename
+	allSpecs, err := ParseSpec(filepath, printf)
+	if err != nil {
+		return err
+	}
+	for _, sequence := range allSpecs.Sequences {
+		for _, node := range sequence.Nodes {
+			if err := check.CheckNode(sequence.Name, *node); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func testFailInvalidValueError(t *testing.T, err error, nodeLevel bool, field, expectedVal, errMsg string) {
```

Comment:
rather than passing in a bunch of stuff, could you just pass in the expected error itself? Same applies to next two funcs.

Reply:
I also don't love that these funcs require the nodes / sequences to be named "node-a" + "seq-a" — it's easy for someone writing a new test not to know that. Passing in the error would help with that.